homeassistant/components/tfl/config_flow.py

```diff
@@ -0,0 +1,230 @@
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except ValueError:
+                errors["base"] = "invalid_stop_point"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                # User input is valid, save the stop point
+                self.data[CONF_STOP_POINTS].append(user_input[CONF_STOP_POINT])
+                # If user ticked the box show this form again so they can add an
+                # additional stop point.
+                if user_input.get(CONF_STOP_POINT_ADD_ANOTHER, False):
+                    return await self.async_step_stop_point()
+
+                # return await self.async_step_stop_points()
+                return self.async_create_entry(
+                    title="Transport for London", data=self.data
+                )
+
+        return self.async_show_form(
+            step_id="stop_point",
+            data_schema=STEP_STOP_POINT_DATA_SCHEMA,
+            errors=errors,
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> config_entries.OptionsFlow:
+        """Create the options flow."""
+        return OptionsFlowHandler(config_entry)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handles options flow for the component."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manage the options for the TfL component."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            updated_stops = user_input[CONF_STOP_POINTS]
+            if (
+                CONF_STOP_POINT in user_input
+                and user_input[CONF_STOP_POINT] is not None
+            ):
+                updated_stops.append(user_input[CONF_STOP_POINT])
+
+            data: dict[str, Any] = {}
+            data[CONF_API_APP_KEY] = user_input[CONF_API_APP_KEY]
+            data[CONF_STOP_POINTS] = updated_stops
+
+            if not errors:
+                # Value of data will be set on the options property of our config_entry
+                # instance.
+                return self.async_create_entry(
+                    title="Transport for London",
+                    data=data,
+                )
+
+        config = config_from_entry(self.config_entry)
+        api_key = deepcopy(config[CONF_API_APP_KEY])
+        all_stops = deepcopy(config[CONF_STOP_POINTS])
+
+        options_schema = vol.Schema(
+            {
+                vol.Optional(CONF_API_APP_KEY, default=api_key): cv.string,
```

Comment:
Why do we allow the user to change the app key from an options flow?
If we can detect the app key is invalidated, we should instead trigger a reauth flow

Reply:
This is because I didn't know there was a `reconfigure` flow.

I'll take a look at it

Reply:
@emontnemery I've actioned this so it now uses a reconfigure flow